cpp/src/strings/convert/convert_integers.cu

```diff
@@ -38,6 +37,147 @@
+ */
+struct dispatch_is_integer_fn {
+  template <typename T, std::enable_if_t<std::is_integral<T>::value>* = nullptr>
+  std::unique_ptr<column> operator()(strings_column_view const& strings,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr) const
+  {
+    auto const d_column = column_device_view::create(strings.parent(), stream);
+    auto results        = make_numeric_column(data_type{type_id::BOOL8},
+                                       strings.size(),
+                                       cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                       strings.null_count(),
+                                       stream,
+                                       mr);
+
+    thrust::transform(rmm::exec_policy(stream),
+                      thrust::make_counting_iterator<size_type>(0),
+                      thrust::make_counting_iterator<size_type>(strings.size()),
+                      results->mutable_view().data<bool>(),
+                      string_to_integer_check_fn<T>{*d_column});
+
+    // Calling mutable_view() on a column invalidates it's null count so we need to set it back
+    results->set_null_count(strings.null_count());
+
+    return results;
+  }
+
+  template <typename T, std::enable_if_t<not std::is_integral<T>::value>* = nullptr>
+  std::unique_ptr<column> operator()(strings_column_view const&,
+                                     rmm::cuda_stream_view,
+                                     rmm::mr::device_memory_resource*) const
+  {
+    CUDF_FAIL("is_integer is expecting an integer type");
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<column> is_integer(
+  strings_column_view const& strings,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+{
+  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto d_column       = *strings_column;
+  // create output column
+  auto results   = make_numeric_column(data_type{type_id::BOOL8},
+                                     strings.size(),
+                                     cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto d_results = results->mutable_view().data<bool>();
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator<size_type>(0),
+                    thrust::make_counting_iterator<size_type>(strings.size()),
+                    d_results,
+                    [d_column] __device__(size_type idx) {
+                      if (d_column.is_null(idx)) return false;
+                      return string::is_integer(d_column.element<string_view>(idx));
+                    });
```

Comment:
```suggestion
  thrust::transform(rmm::exec_policy(stream),
                    strings_column.pair_begin(),
                    strings_column.pair_end(),
                    d_results,
                    [d_column] __device__(auto p) {
                      return p.second ? is_integer(p.first) : false;
                    });
```

Reply:
This becomes even nicer when we replace `pair_iterator` with `optional_iterator` https://github.com/rapidsai/cudf/issues/6470

Reply:
The reason I don't like this is because the pair iterator requires the `has_nulls` at compile time and thus this doubles the compile time. This looks nice and all but I'd rather we waited until the pair iterator was changed to make it a runtime parameter instead.

Reply:
What is blocking that change?

Reply:
@davidwendt please request changes when something is a problem otherwise the problem will get merged.

Reply:
The main concern here is something like this:
```
if (strings.has_nulls()) {
    thrust::transform(
      rmm::exec_policy(stream),
      d_column->pair_begin<string_view, true>(),
      d_column->pair_end<string_view, true>(),
      d_results,
      [] __device__(auto const& p) { return p.second ? string::is_integer(p.first) : false; });
  } else {
    thrust::transform(
      rmm::exec_policy(stream),
      d_column->pair_begin<string_view, false>(),
      d_column->pair_end<string_view, false>(),
      d_results,
      [] __device__(auto const& p) { return p.second ? string::is_integer(p.first) : false; });
  }
```

So we will have 2 calls to `thrust::transform` instead of just one. I can verify that this makes the compile time increases by less than 1 second.

Reply:
Before:
```
6:07 PM	Build finished in 25 sec, 27 ms
```

After:
```

6:08 PM	Build finished in 26 sec, 241 ms
```